src/cmap/wire_protocol/on_demand/document.ts

```diff
@@ -0,0 +1,296 @@
+import {
+  Binary,
+  BSON,
+  type BSONElement,
+  BSONError,
+  type BSONSerializeOptions,
+  BSONType,
+  getBigInt64LE,
+  getFloat64LE,
+  getInt32LE,
+  ObjectId,
+  parseToElementsToArray,
+  Timestamp,
+  toUTF8
+} from '../../../bson';
+import { StringFinder } from './string_finder';
+
+// eslint-disable-next-line no-restricted-syntax
+const enum BSONElementOffset {
+  type = 0,
+  nameOffset = 1,
+  nameLength = 2,
+  offset = 3,
+  length = 4
+}
+
+export type JSTypeOf = {
+  [BSONType.int]: number;
+  [BSONType.long]: bigint;
+  [BSONType.timestamp]: Timestamp;
+  [BSONType.binData]: Binary;
+  [BSONType.bool]: boolean;
+  [BSONType.objectId]: ObjectId;
+  [BSONType.string]: string;
+  [BSONType.date]: Date;
+  [BSONType.object]: OnDemandDocument;
```

Comment:
We kind of assume that the server would omit keys rather than set them to null or undefined. What / how should support for those values look like if at all?

Reply:
This is no longer relevant, right?

Reply:
Yes, resolved!